modules/auxiliary/scanner/http/git_scanner.rb

```diff
@@ -0,0 +1,108 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::WmapScanServer
+  include Msf::Auxiliary::Scanner
+  include Msf::Auxiliary::Report
+
+  def initialize
+    super(
+      'Name'        => 'HTTP Git Scanner',
+      'Description' => 'Detect Git disclosure.',
+      'Author'      => ['Nixawk'],
+      'References'  => [
+        ['URL', 'https://github.com/git/git/blob/master/Documentation/technical/index-format.txt']
+      ],
+      'License'     => MSF_LICENSE
+    )
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The test path to .git directory', '/.git/']),
+        OptBool.new('GIT_INDEX', [true, 'Check index file in .git directory', true]),
+        OptBool.new('GIT_CONFIG', [false, 'Check config file in .git directory', true]),
+        OptBool.new('GIT_HEAD', [false, 'Check HEAD file in .git directory', true])
+      ]
+    )
+  end
+
+  def req(filename)
+    send_request_cgi(
+      'uri' => normalize_uri(target_uri, filename)
+    )
+  end
+
+  def git_index_parse(resp)
+    return if resp.blank? || resp.length < 12 # A 12-byte header
+    signature = resp[0, 4]
+    return unless signature == 'DIRC'
+
+    version = resp[4, 4].unpack('N')[0].to_i
+    entries_count = resp[8, 4].unpack('N')[0].to_i
+
+    return unless version && entries_count
+    print_good("#{full_uri} (git repo version #{version}) - #{entries_count} files found")
+
+    report_note(
+      host: rhost,
+      port: rport,
+      proto: 'tcp',
+      type: 'git_disclosure',
+      data: { full_uri: full_uri, version: version, entries_count: entries_count }
```

Comment:
My vote on this this no -- somewhat regardless of how much information you provide to the user in the form of loot, notes, etc, chances are if they are going to do much with the knowledge gained by the running of this module, that will absolutely require that they actually get a copy of the repository in question and examine the _contents_ of the files, for example.  

For super simple repositories, the overhead of storing the information relative to the value it provides is OK, but when you are talking about much larger repositories, I think very quickly the value of anything beyond the fact that _"yes, there is a repository here" is extra information; both the count and names of the files in this repository are useful bits of information, but with that alone there is little to be gained.